packages/language-support/src/antlr-grammar/CypherParser.g4

```diff
@@ -651,10 +655,10 @@ propertyKeyName
    ;
 
 parameter[String paramType]
-   : DOLLAR parameterName
+   : DOLLAR parameterName[paramType]
    ;
 
-parameterName
+parameterName[String paramType]
```

Comment:
I think it can be any string actually. I think we have some logic in the auto-completions to filter what type of parameters (normally we just want string parameters) to offer depending on the rule, so maybe we can use this instead. I'll card it!